### PR TITLE
Add eslint and prettier formatting

### LINKED ISSUE #2259: Add linting and formatting to frontend JS
To help maintain and standardized frontend JS development, code linting and formatting should be added to the testing/build process.

[eslint](https://eslint.org/) and [prettier](https://prettier.io/) are two popular projects used to lint and format respectively. Adding their functionality to the NPM scripts with a git pre-commit hook would be most advantageous. NPM specific Git hooks could be setup with something like [ghooks](https://www.npmjs.com/package/ghooks)

### PR DESCRIPTION
**Pull request by apburnes**
## Summary

Add eslint and prettier formatting to the frontend JS to improve code quality and maintenance.

- Resolves #2259 
_Added eslint and prettier node packages and ran scripts to format all of the frontend JS code. This commit adds the packages, the npm scripts `npm run lint`, `npm run format`, and `npm run format-lint`, and `npm run format-lint` has been run to fix any linting/formatting issues in the current JS_

## Impacted areas of the application
- All JS code within `fec/fec/static/js`

### CODE REVIEW

File: fec/fec/static/js/legal/SearchResults.js [hunk 1]
```diff
@@ -2,78 +2,162 @@ const React = require('react');
 const $ = require('jquery');
 const moment = require('moment');
 
-
 function SearchResults(props) {
-  function highlights(advisory_opinion) {
-    return {__html: '&hellip;' + advisory_opinion.highlights }
-  }
+  // function highlights(advisory_opinion) {
```
Comment (lbeaufort) [inline fec/fec/static/js/legal/SearchResults.js]: @apburnes did you mean to comment out this function?

File: fec/fec/static/js/legal/SearchResults.js [hunk 2]
```diff
@@ -2,78 +2,162 @@ const React = require('react');
...
+                                  {citation.no}
+                                </a>
+                              </div>
+                            );
+                          })
+                        : 'This advisory opinion is not cited by other advisory opinions'}
+                    </td>
+                  </tr>
+                );
+                // {
```
Comment (lbeaufort) [inline fec/fec/static/js/legal/SearchResults.js]: same as above - not sure this was intended to be commented out

**Comment by patphongs** [comment]
@apburnes Can you please rebase from develop?

**Comment by apburnes** [comment]
- [x]  Added and now using the recommend rules from `eslint` and `eslint-plugin-react`
  - [eslint list of recommended rules](https://eslint.org/docs/rules/)
  - [eslint-plugin-react list of recommended rules](https://github.com/yannickcr/eslint-plugin-react#list-of-supported-rules)

- [x] Added the linting script `npm run lint` to `npm run test-single` so the linter would be run in CircleCI.  It will fail the build if the lint fails.

- [x] New linting and formatting capabilities are available
  - `npm run format` formats all frontend JS code
  - `npm run lint` lints all frontend JS code (except `/vendor` folder)
  - `npm run format-lint` will format and then lint all JS code
  - `npm run test-single` linting added before running tests

File: fec/fec/static/js/legal/SearchResults.js (hunk 2 above)
Comment (apburnes) [reply fec/fec/static/js/legal/SearchResults.js]: Linter identified as unreachable code so will comment out for further review.

**Pull request merged**